valohai_yaml/objs/parameter.py

```diff
@@ -62,6 +64,8 @@ def __init__(
         self.choices = (list(choices) if choices else None)
         self.multiple = MultipleMode.cast(multiple)
         self.multiple_separator = str(multiple_separator or ',')
+        self.widget = ParameterWidget.parse(widget)
```

Comment:
Just
```suggestion
        self.widget = widget
```
here, and call `ParameterWidget.parse` in `Parameter`'s `parse`, á la

```python
    @classmethod
    def parse(cls, data: SerializedDict) -> 'Parameter':
        kwargs = data.copy()
        kwargs['widget'] = ParameterWidget.parse(data.get('widget'))
        return cls(**kwargs)
```

Reply:
Sorry! I was rushing it and had a lukihäiriö. 

It should be kosher now.